src/test/java/teammates/test/cases/datatransfer/InstructorAttributesTest.java

```diff
@@ -18,69 +18,92 @@
 public class InstructorAttributesTest extends BaseTestCase {
 
     @Test
-    public void testConstructor() {
-        @SuppressWarnings("deprecation")
-        InstructorAttributes instructor =
-                new InstructorAttributes("valid.google.id", "valid-course-id", "valid name", "valid@email.com");
+    public void testBuilder() {
+        InstructorAttributes instructor = InstructorAttributes
+                .builder("valid.google.id", "valid-course-id", "valid name", "valid@email.com")
+                .build();
         String roleName = Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER;
         String displayedName = InstructorAttributes.DEFAULT_DISPLAY_NAME;
         InstructorPrivileges privileges =
                 new InstructorPrivileges(Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER);
 
+        // Check default values for optional params
         assertEquals(roleName, instructor.role);
         assertEquals(displayedName, instructor.displayedName);
         assertEquals(privileges, instructor.privileges);
+        assertFalse(instructor.isArchived);
+        assertTrue(instructor.isDisplayedToStudents);
 
-        InstructorAttributes instructor1 =
-                new InstructorAttributes(instructor.googleId, instructor.courseId, instructor.name, instructor.email,
-                                         instructor.role, instructor.displayedName,
-                                         instructor.getTextFromInstructorPrivileges());
+        InstructorAttributes instructor1 = InstructorAttributes
```

Comment:
Your `instructor1` `instructor2` `instructorNew` just copy everything from `instructor`. However, many fields of `instructor` is default values. So you don't know whether the builder builds it correctly by copying the correct fields or just the input value is invalid and the builder chooses the default value.